src/en/mangaplex/src/eu/kanade/tachiyomi/extension/en/mangaplex/MangaPlex.kt

```diff
@@ -0,0 +1,124 @@
+
+    override fun latestUpdatesSelector() = "#posts-container li:has(.post-details)"
+
+    override fun latestUpdatesFromElement(element: Element): SManga {
+        val manga = SManga.create()
+
+        manga.thumbnail_url = element.select(".post-thumb img").attr("src")
+        // using search for manga page and chapter list
+        manga.url = element.select("h3.post-title a").attr("href").substringBeforeLast("-chapter").replace("$baseUrl/", "/search/")
+        val mangaTitleSelector = element.select(".post-details p.post-excerpt").text().substringAfter("Read ").substringBefore(" Chapter")
+        manga.title =
+            if (mangaTitleSelector.contains("manga", true) || mangaTitleSelector.contains("manhwa", true) || mangaTitleSelector.contains("manhua", true)) mangaTitleSelector.substringBeforeLast(" ")
+            else mangaTitleSelector
+        return manga
+    }
+
+    // popular
+    override fun popularMangaRequest(page: Int): Request = latestUpdatesRequest(page)
+
+    override fun popularMangaSelector() = latestUpdatesSelector()
+
+    override fun popularMangaFromElement(element: Element): SManga = latestUpdatesFromElement(element)
+
+    override fun popularMangaNextPageSelector(): String = latestUpdatesNextPageSelector()
+
+    // search
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return GET("$baseUrl/search/$query/page/$page")
+    }
+
+    override fun searchMangaSelector() = "#posts-container li:has(h5:contains(Manga)):has(p:contains(Chapter))"
+
+    override fun searchMangaFromElement(element: Element): SManga = latestUpdatesFromElement(element)
+
+    override fun searchMangaNextPageSelector(): String = latestUpdatesNextPageSelector()
+
+    // manga details
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply { /*empty*/ }
+
+    // chapters
+    override fun chapterListRequest(manga: SManga) = chapterListRequest(manga.url, 1)
+
+    private fun chapterListRequest(mangaUrl: String, page: Int): Request {
+        val mangaUrlClean = if (mangaUrl.contains(baseUrl)) mangaUrl.substringAfter(baseUrl) else mangaUrl
+        return GET("$baseUrl$mangaUrlClean/page/$page", headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        var document = response.asJsoup()
+        val chapters = mutableListOf<SChapter>()
+        var nextPage = 2
+        document.select(chapterListSelector()).map { chapters.add(chapterFromElement(it)) }
+        while (document.select(paginationNextPageSelector).isNotEmpty()) {
+            val currentPage = document.select("meta[property=\"og:url\"]").attr("content")
+            document = client.newCall(chapterListRequest(currentPage, nextPage)).execute().asJsoup()
+            document.select(chapterListSelector()).map { chapters.add(chapterFromElement(it)) }
+            nextPage++
+        }
+
+        return chapters
+    }
```

Comment:
```suggestion
    override fun chapterListParse(response: Response): List<SChapter> {
        var document = response.asJsoup()
        val chapters = document.select(chapterListSelector())
            .map(::chapterFromElement)
            .toMutableList()
        var nextPage = 2
        
        while (document.select(paginationNextPageSelector).isNotEmpty()) {
            val currentPage = document.select("meta[property=\"og:url\"]").attr("content")
            document = client.newCall(chapterListRequest(currentPage, nextPage)).execute().asJsoup()
            chapters += document.select(chapterListSelector())
                .map(::chapterFromElement)
            nextPage++
        }

        return chapters
    }
```

Please test it, I am not sure if `chapters` should be `val` or `var` when using `+=`.